### PR TITLE
Create campaign controller

### PR DESCRIPTION
**Pull request by rjvani**
Add GET, POST, and PUT endpoints for `Campaign`s

### CODE REVIEW

File: spec/requests/campaigns_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      end
+
+      it 'Returns 200' do
+        expect(response).to have_http_status(200)
+      end
+    end
+  end
+
+  context 'POST /campaigns' do
+    context 'With invalid parameters' do
```
Comment (justintmckibben) [inline spec/requests/campaigns_request_spec.rb:51]: Should we test invalid seller_id and location_id?

File: spec/factories/campaign.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :campaign do
+    active { true }
+    valid { true }
+    end_date { DateTime.now }
```
Comment (justintmckibben) [inline spec/factories/campaign.rb:7]: Faker has a way to generate a random date in the future: `Faker::Time.forward(days: 30)   # any time up to 30 days into the future`

File: spec/requests/campaigns_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+          '/campaigns',
+          params: {
+            end_date: Date.tomorrow,
+            location_id: @location.id,
+            seller_id: @seller.seller_id,
+          },
+          as: :json,
+        )
+      end
+
```
Comment (justintmckibben) [inline spec/requests/campaigns_request_spec.rb:71]: could you add a test that Campaigns.all.size is 1? and maybe has the attributes you passed in

File: app/controllers/campaigns_controller.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class CampaignsController < ApplicationController
+  before_action :set_campaign, only: %i[show update]
+
+  # GET /campaigns
+  def index
+    json_response(Campaign.all)
```
Comment (justintmckibben) [inline app/controllers/campaigns_controller.rb:8]: We should Order by end_date where the one that is ending soonest is the first

File: spec/requests/campaigns_request_spec.rb (hunk 1 above)
Comment (rjvani) [reply spec/requests/campaigns_request_spec.rb:51]: Good call, added

File: spec/requests/campaigns_request_spec.rb (hunk 3 above)
Comment (rjvani) [reply spec/requests/campaigns_request_spec.rb:127]: Added a test to look up the new record and verify attributes

File: app/controllers/campaigns_controller.rb (hunk 4 above)
Comment (rjvani) [reply app/controllers/campaigns_controller.rb]: Added the order, but would we need to filter on `active` also?

Comment (rjvani) [reply app/controllers/campaigns_controller.rb]: Added a query parameter for this endpoint

**Pull request merged**